app/models/post_flag.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+class PostFlag < ActiveRecord::Base
+  scope :enabled, -> { where(enabled: true) }
+
+  before_create :set_custom_id, unless: :system?
+  before_save :set_position
+  after_save :reset_post_action_types!
+  after_destroy :reset_post_action_types!
+
+  def used?
+    PostAction.exists?(post_action_type_id: self.id)
+  end
+
+  def self.reset_post_action_types!
+    PostActionType.initialize_flag_settings
+  end
+
+  private
+
+  def reset_post_action_types!
+    self.class.reset_post_action_types!
+  end
+
+  def set_custom_id
+    return if self.id.to_i > 1000
```

Comment:
Is this to give space after the system ones? Instead, you may want to set the sequence in the initial database migration, after the system flags have been seeded. You could do `SELECT setval('flags_id_seq', 1001, FALSE);` https://www.postgresql.org/docs/16/functions-sequence.html